### ISSUE TITLE
Pre-training details about PPIRef

### ISSUE THREAD

**Issue by mylRalph** [anton-bushuiev/PPIformer#2]
Hi, Congrats on your excellent work! 

I try to pretrain my model on PPIRef50K. However, it seems that there is no specific description of the data-splitting strategy in your paper. How do you split the PPIRef50K for pre-training, conduct the validation, and select the pre-training checkpoint? Besides, given several protein-protein interfaces, how do you construct batches for training? Specifically, do you crop interfaces into patches and conduct padding like RDE? I would really appreciate it if I could get your help! Thank you!

Regards,
Ralph

**Comment by anton-bushuiev** [comment]
Hi, Ralph,

Thank you for your comment. It should be explained more clearly in the camera-ready version of the paper.

> How do you split the PPIRef50K for pre-training, conduct the validation, and select the pre-training checkpoint?

We pre-train on the whole PPIRef50K and monitor zero-shot ddG performance on the training part of SKEMPI for validation/selecting checkpoints. 

Specifically, paths to the PPIs for training can be read via `ppiref` as following:
```
from ppiref.split import read_fold
ppi_paths_pre_training = read_fold('ppiref_6A_filtered_clustered_04', 'whole')
```
The `ppiref_6A_filtered_clustered_04` contains only a single fold `whole` (and not for example two folds `training` and `validation`) because we did not split PPIRef50K. Instead, for validation we used the PPIs from SKEMPI and measured zero-shot ddG to select the best pre-training checkpoints. You can read all validation PPIs as below:
```
ppi_paths_fine_tuning = read_fold('skempi2_iclr24_split', '0+1+2')
```
The paths in the `ppi_paths_fine_tuning` list correpond to the interfaces extracted from the cleaned .pdb files available on [the SKEMPI website](https://life.bsc.es/pid/skempi2/database/index). Finally, we use these 3 folds (`0`, `1`, `2`) to fine-tune three models starting from the same pre-trained checkpoint (using for example `0+1` for training and `2` for validation), and test the ensamble of three models (averaging the predictions) on the SKEMPI test set:
```
ppi_paths_test = read_fold('skempi2_iclr24_split', 'test')
```

> Besides, given several protein-protein interfaces, how do you construct batches for training? Specifically, do you crop interfaces into patches and conduct padding like RDE?

For a batch, for example, of size 2, we use 2 complete interfaces. Then we mask some ratio of nodes (50% in the final model) from each interface, such that only the amino acids from a single partner in the interface are masked. So, we do not crop PPIs into patches. Technically, we collate several PPIs into a batch using Pytorch Geometric, which is described in detail on [this page in the PyG docs](https://pytorch-geometric.readthedocs.io/en/latest/advanced/batching.html).

Notes:
- For pre-training we used PPIRef50K with the interfaces based on 10A distance between heavy atoms. Now, on PPIRef Zenodo there are only interfaces based on 6A distance. I will upload the 10A interfaces, updated to this year, in the nearest future. Nevertheless, at the moment I do not know if training from 6A interfaces will lead to better or worse performance. Both distance are well-established in literature.
- If you liked to split PPIRef50K, I would suggest using the pre-computed iDist embeddings for PPIRef300K available on PPIRef Zenodo.

Best,
Anton

**Comment by mylRalph** [comment]
Hi, Anton,

Thanks so much for your detailed clarification! It did help a lot. Regarding splitting PPIRef50K, do you have any recommendations for the threshold when using the pre-computed iDist embeddings? Since PPIRef50K has been deduplicated under the threshold of 0.04, so I guess maybe random splitting into training and validation parts is feasible, too?

Regards,
Ralph

**Comment by anton-bushuiev** [comment]
Hi, Ralph.

Yes, you are right. Splitting PPIRef50K randmly is reasonable. Similarly, to randomly splitting UniRef seeds.

Nevertheless, it is potentially possible to achieve an even more struct split by, for example, connecting PPIRef50K using a higher threshold (such as 0.06) and then splitting the connected components of PPIs.

Best,
Anton
